lib/cdo/rack/whitelist.rb

```diff
@@ -0,0 +1,108 @@
+# Rack middleware that whitelists cookies and headers based on path-based cache behaviors.
+# Behaviors are defined in http cache config.
+require_relative '../../../cookbooks/cdo-varnish/libraries/helpers'
+require 'active_support/core_ext/hash/slice'
+require 'cdo/rack/response'
+
+module Rack
+  module Whitelist
+    # Downstream middleware filters out unwanted HTTP request headers and cookies,
+    # and extracts cookies into HTTP headers before the request reaches the cache.
+    class Downstream
+      attr_reader :config
+
+      def initialize(app, config)
+        @app = app
+        @config = config
+      end
+
+      REQUEST_METHODS = %w(HEAD DELETE POST GET OPTIONS PUT PATCH)
```

Comment:
Please comment that these are chosen to match the supported verbs in Cloudfront.